SimDataFormats/HTXS/interface/HiggsTemplateCrossSections.h

```diff
@@ -0,0 +1,238 @@
+      HTXS::Stage0::Category stage0_cat;
+      HTXS::Stage1::Category stage1_cat_pTjet25GeV;
+      HTXS::Stage1::Category stage1_cat_pTjet30GeV;
+      // Error code :: classification was succesful or some error occured
+      HTXS::ErrorCode errorCode;
+    };
+    
+    template <class category>
+      inline HTXS::HiggsClassification Rivet2Root(category const &htxs_cat_rivet){
+      HTXS::HiggsClassification cat;
+      cat.prodMode = htxs_cat_rivet.prodMode;
+      cat.errorCode = htxs_cat_rivet.errorCode;
+      cat.higgs = MakeTLV(htxs_cat_rivet.higgs);
+      cat.V = MakeTLV(htxs_cat_rivet.V);
+      cat.p4decay_higgs = MakeTLV(htxs_cat_rivet.p4decay_higgs);
+      cat.p4decay_V = MakeTLV(htxs_cat_rivet.p4decay_V);
+      cat.jets25 = MakeTLVs(htxs_cat_rivet.jets25);
+      cat.jets30 = MakeTLVs(htxs_cat_rivet.jets30);
+      cat.stage0_cat = htxs_cat_rivet.stage0_cat;
+      cat.stage1_cat_pTjet25GeV = htxs_cat_rivet.stage1_cat_pTjet25GeV;
+      cat.stage1_cat_pTjet30GeV = htxs_cat_rivet.stage1_cat_pTjet30GeV;
+      return cat;    
+    }
+    
+
+    
+    inline int HTXSstage1_to_HTXSstage1FineIndex(HTXS::Stage1::Category stage1, 
+						 HiggsProdMode prodMode, tH_type tH) {
+
+      if(stage1==HTXS::Stage1::Category::UNKNOWN) return 0;
+      int P = (int)(stage1 / 100);
+      int F = (int)(stage1 % 100);
+      // 1.a spit tH categories
+      if (prodMode==HiggsProdMode::TH) {
+	// check that tH splitting is valid for Stage-1 FineIndex
+	// else return unknown category
+	if(tH==tH_type::noTH) return 0;
+	// check if forward tH
+	int fwdH = F==0?0:1;
+	return (49 + 2*(tH-1) +fwdH);
+      }
+      // 1.b QQ2HQQ --> split into VBF, WH, ZH -> HQQ
+      // offset vector 1: input is the Higgs prodMode 
+      // first two indicies are dummies, given that this is only called for prodMode=2,3,4 
+      std::vector<int> pMode_offset = {0,0,13,19,25};
+      if (P==2) return (F + pMode_offset[prodMode]);
+      // 1.c remaining categories
+      // offset vector 2: input is the Stage-1 category P 
+      // third index is dummy, given that this is called for category P=0,1,3,4,5,6,7
+      std::vector<int> catP_offset = {0,1,0,31,36,41,45,47};
+      return (F + catP_offset[P]);
+    }
+
+    inline int HTXSstage1_to_HTXSstage1FineIndex(const HiggsClassification &stxs, 
+						 tH_type tH=noTH, bool jets_pT25 = false) {
+      HTXS::Stage1::Category stage1 = 
+	jets_pT25==false?stxs.stage1_cat_pTjet30GeV:
+	stxs.stage1_cat_pTjet25GeV;
+      return HTXSstage1_to_HTXSstage1FineIndex(stage1,stxs.prodMode,tH);
+    }
+    
+    inline int HTXSstage1_to_index(HTXS::Stage1::Category stage1) {
+      // the Stage-1 categories
+      int P = (int)(stage1 / 100);
+      int F = (int)(stage1 % 100);    
+      std::vector<int> offset{0,1,13,19,24,29,33,35,37,39};
+      // convert to linear values
+      return ( F + offset[P] );
+    }
+
+     
+// #endif 
+
+} // namespace HTXS
+
+
+#ifdef RIVET_Particle_HH
```

Comment:
go ahead and start a new file for the new class.

Reply:
why whold that need a separate .h ?  (please note the ifdef is not a ifndef protecting a file )

Reply:
belated answer - to follow a cms convention that makes it easier to find headers. Anyway, deal with it as a followup pr